Source/BugsnagConfiguration.m

```diff
@@ -170,13 +170,28 @@ - (nonnull id)copyWithZone:(nullable NSZone *)zone {
  * @returns A boolean representing whether the apiKey is valid.
  */
 + (BOOL)isValidApiKey:(NSString *)apiKey {
+    if ([self isApiKeyMissing:apiKey]) {
+        return false;
+    }
+
     NSCharacterSet *chars = [[NSCharacterSet
         characterSetWithCharactersInString:@"0123456789ABCDEF"] invertedSet];
-    BOOL isHex = [apiKey isKindOfClass:[NSString class]] &&
-            (NSNotFound == [[apiKey uppercaseString] rangeOfCharacterFromSet:chars].location);
+
+    BOOL isHex = (NSNotFound == [[apiKey uppercaseString] rangeOfCharacterFromSet:chars].location);
+
     return isHex && [apiKey length] == BSGApiKeyLength;
 }
 
+/**
+ * Check if the API Key is missing, i.e. it's nil, not a string or is empty
+ *
+ * This is distinct from 'isValidApiKey' to allow throwing an exception in this case, but not
+ * if the API Key is in the wrong format
+ */
++ (BOOL)isApiKeyMissing:(NSString *)apiKey {
```

Comment:
This method would benefit from @param/@return documentation.  Also, is the `isKindOfClass` comparison necessary?  Users are warned about incorrect assignment and incorrect type assignment is not possible in Swift.  I wonder if it's overkill.

Reply:
The `[apiKey length]` call caused one or two tests to throw an exception because they passed the wrong type, when they were expecting `isValidApiKey` to return false safely. This just moved from where `isHex` is set rather than being a new check, but I can remove it if you'd prefer it to throw when given an invalid type — `setApiKey` will throw either way, but `isValidApiKey` is used in a few other places too

Reply:
I went through the rest of the code and revised my original opinion on type checking; I think there is a case for it.  I edited the comment, I suspect while you were replying.  The doc nit still stands.